From 6f06cac5f6b980aae048f9543e9af2cb7e46b69b Mon Sep 17 00:00:00 2001
From: Liam Donovan <liam@liamdonovan.co.uk>
Date: Tue, 29 May 2018 18:10:45 +0100
Subject: [PATCH] bela bootcmd

---
 include/configs/am335x_evm.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index 5582a66..135d1e2 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -231,7 +231,12 @@
 	NANDARGS \
 	NETARGS \
 	DFUARGS \
-	BOOTENV
+	BOOTENV \
+	"bootcmd="	\
+		"if load mmc 0:1 ${loadaddr} uEnv.txt; then echo \"booting from SD card\"; else load mmc 1:1 ${loadaddr} uEnv.txt; echo \"booting from eMMC\"; fi;"	\
+		"env import -t ${loadaddr} ${filesize};"	\
+		"run findfdt;"	\
+		"run uenvcmd;\0"
 #endif
 
 /* NS16550 Configuration */
-- 
2.1.4

