From 88ddfcc51e0d050a0d2ad8a2f173f67d17d0cad5 Mon Sep 17 00:00:00 2001
From: Liam Donovan <liam@liamdonovan.co.uk>
Date: Mon, 28 May 2018 22:17:56 +0100
Subject: [PATCH] bela uboot overlays

---
 include/configs/am335x_evm.h | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index fbc9d76..148d4de 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -236,7 +236,14 @@
 		"if load mmc 0:1 ${loadaddr} uEnv.txt; then echo \"booting from SD card\"; else load mmc 1:1 ${loadaddr} uEnv.txt; echo \"booting from eMMC\"; fi;"	\
 		"env import -t ${loadaddr} ${filesize};"	\
 		"run findfdt;"	\
-		"run uenvcmd;"
+		"run uenvcmd; \0"	\
+	"bela_loadoverlay="	\
+		"echo loading ${overlay};"	\
+		"load mmc ${mmcid}:2 ${rdaddr} ${overlay};"	\
+		"fdt addr $fdtaddr;"	\
+		"fdt resize 0x60000;"	\
+		"fdt apply ${rdaddr};"	\
+		"fdt resize 0x60000;"
 
 #endif
 
-- 
2.1.4

